src/components/App/App.test.jsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import App from './App.component';
+
+describe('App Component Tests', () => {
+  beforeEach(() => {
+    render(<App />);
+  });
+
+  it('Should render the Header component', async () => {
+    expect(document.querySelector('header')).toBeInTheDocument();
+  });
+
+  it('Should render the Home component', async () => {
+    expect(screen.getByTestId('home-component')).toBeInTheDocument();
+  });
+
+  it('Should change background color with a click', async () => {
+    const { body } = document;
+    const spy = jest.spyOn(body.style, 'setProperty');
+
+    body.click();
+    expect(spy).toHaveBeenCalled();
```

Comment:
I couldn't figure out what is being tested here, can you exaplain me?

Reply:
Yes, this was to test that the useEffect in App component had been called. The one that changes the colour background